### PR TITLE
feat: add grpc support for log autoexport

### PR DESCRIPTION
**Pull request by sysulq**

### CODE REVIEW

File: exporters/autoexport/logs_test.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ func TestLogExporterOTLPWithDedicatedProtocol(t *testing.T) {
 		protocol, clientType string
 	}{
 		{"http/protobuf", "atomic.Pointer[go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp.client]"},
+		{"grpc", "otlploggrpc.logClient"},
```
Comment (pellared) [inline exporters/autoexport/logs_test.go:70]: Can we add the same for `TestLogExporterOTLP`?

File: exporters/autoexport/logs.go [hunk 2]
```diff
@@ -31,6 +32,8 @@ var logsSignal = newSignal[log.Exporter]("OTEL_LOGS_EXPORTER")
 // supported values:
 //   - "http/protobuf" (default) -  protobuf-encoded data over HTTP connection;
 //     see: [go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp]
+//   - "grpc" - gRPC with protobuf-encoded data over HTTP/2 connection;
```
Comment (pellared) [inline exporters/autoexport/logs.go:35]: Can you add an entry to changelog?

**Review by dmathieu — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Added
 
 - Add `NewProducer` to `go.opentelemetry.io/contrib/instrumentation/runtime`, which allows collecting the `go.schedule.duration` histogram metric from the Go runtime. (#5991)
+- Add grpc support for log autoexport in `go.opentelemetry.io/contrib/exporters/autoexport`. (#6083)
```
Comment (pellared) [inline CHANGELOG.md:14]:
```suggestion
- Add gRPC protocol support for OTLP log exporter in `go.opentelemetry.io/contrib/exporters/autoexport`. (#6083)
```

**Comment by pellared** [comment]
@sysulq, thanks for your contribution.

**Pull request merged**